### ISSUE TITLE
Unable to remove node from cluster without stopping it

### ISSUE THREAD

**Issue by andrehp** [erlio/vernemq#82]
Hello,

I'm doing some tests with clusters and to do so I have two instances of VerneMQ running on different machines.

I was able to put them on the same cluster and test it is working properly, but I wasn't able to remove one node from the cluster correctly. The two machines are VerneMQChat@10.0.1.76 and VerneMQChat@10.0.1.23.

From .76 I ran
`sudo vmq-admin cluster join discovery-node=VerneMQChat@10.0.1.23`
and the result was as expected.
```
+---------------------+-------+
|        Node         |Running|
+---------------------+-------+
|VerneMQChat@10.0.1.23| true  |
|VerneMQChat@10.0.1.76| true  |
+---------------------+-------+
```

But then I tried to remove the node .23 from the cluster from .76:
```
ubuntu@ip-10-0-1-76:~$ sudo vmq-admin cluster leave node=VerneMQChat@10.0.1.23
Done! Use -k to teardown and migrate existing sessions!
ubuntu@ip-10-0-1-76:~$ sudo vmq-admin cluster leave node=VerneMQChat@10.0.1.23 -k
RPC to 'VerneMQChat@10.0.1.76' failed: timeout
```

The result of status was still
```
+---------------------+-------+
|        Node         |Running|
+---------------------+-------+
|VerneMQChat@10.0.1.23| true  |
|VerneMQChat@10.0.1.76| true  |
+---------------------+-------+
```

Only when I stopped the instance on .23 by hand (killing it because stop isn't working) is that I got the output
```
+---------------------+-------+
|        Node         |Running|
+---------------------+-------+
|VerneMQChat@10.0.1.23| false |
|VerneMQChat@10.0.1.76| true  |
+---------------------+-------+
```

And after running leave node again I got an error and then .23 disappeared from the list. But when I run vmq-admin cluster status on .76 (after restarting it) it still shows both nodes as active in the cluster.

I then tried adding .23 back to the cluster, the command response was "Done!" but status doesn't reflect that and console.log has the output as follows:
`2016-01-29 19:15:31.647 [warning] <0.7428.0>@vmq_cluster_com:loop:70 terminate due to remote_node_not_available`

Also, I wasn't able to find information about the -k flag for cluster leave. Does it migrate offline messages and queues to one of the other servers? I would really appreciate a way of removing a node from the cluster and still having all the information on one of the other nodes, particularly the messages queued to offline users.

**Comment by dergraf** [comment]
Hi there!

Sorry for the late response! We hope to be able to serve our community better in the future! Please let us know if we can send you any Stickers, if that helps ;)

We are about to release a new version 0.12.5, which addresses the issues you've reported. Please have a look at the new version, and provide us with your valuable feedback!

Thanks a ton!

Best,
Andre and all the VerneMQ Mission Engineers!

**Comment by dergraf** [comment]
Have you had a chance to look at the new release 0.12.5p1?

**Comment by andrehp** [comment]
Hi,

Thanks for the update. I still haven't had the time to look at the updated version.

Would a server with 0.12.5 work properly in a cluster with 0.12.4?

**Comment by dergraf** [comment]
No, unfortunately not. 0.12.4 requires certain rpc calls, which aren't implemented in 0.12.5.

**Comment by andrehp** [comment]
But can I upgrade a server from 0.12.4 to 0.12.5 without major trouble?

**Comment by dergraf** [comment]
Yes you can.

**Comment by andrehp** [comment]
Ok, even with version 0.12.5p2 I'm still unable to remove a node from a cluster.

I run
`sudo vmq-admin cluster leave node=VerneMQ@10.0.0.26`
`sudo vmq-admin cluster leave node=VerneMQ@10.0.0.26 -k`

And the node isn't removed from the cluster. I tried running leave -k multiple times to no effect. The log shows after running the -k command. By the way, there is no effect in the log of running the first command (without -k).

`
2016-04-12 17:54:10.930 [info] <0.3242.0>@vmq_server_cli:wait_till_all_offline:258 QUEUE SUMMARY: 1 online, 0 wait_for_offline, 0 draining, 236 offline, 5 msgs
`

By the way, neither -t nor -i work for the cluster leave command.

**Comment by ioolkos** [comment]
Thanks for reporting! 
@dergraf need your help on this. Have I documented this wrongly? -t and -i indeed seem to not be there.

**Comment by andrehp** [comment]
More information: when I run `vmq-admin listener show` and the mqtt listener is stopped as it should be.

**Comment by dergraf** [comment]
@ioolkos they are in the code https://github.com/erlio/vmq_server/blob/master/src/vmq_server_cli.erl#L122-L131 ;)

**Comment by ioolkos** [comment]
@dergraf sorry, wrongly worded. the flags are there. the point was: they don't seem to work.

**Comment by dergraf** [comment]
https://github.com/erlio/vernemq/issues/126

**Comment by larshesel** [comment]
Hi @andrehp the cluster leave issues are fixed in the latest release (0.12.5p4). If you're still experiencing any issues please reopen or file a new issue.
